widgetastic_manageiq/__init__.py

```diff
@@ -1022,6 +1022,48 @@ def workaround_save_issue(self):
         )
 
 
+class ReactCodemirror(Widget):
+    """Looks the same as ScriptBox but it is changed in 5.11
+    https://github.com/JedWatson/react-codemirror"""
+
+    ROOT = "//div[contains(@class,'miq-codemirror')]//textarea"
+    PARENT_DIV = "./.."  # we need to change visibility of this one
+    CLICK_DIV = "./../.."  # the widget has very nested divs,we need to click on that one ¯\_(ツ)_/¯
+
+    def __init__(self, parent, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+
+    def __locator__(self):
+        return self.ROOT
+
+    @property
+    def element(self):
+        return self.browser.element(self)
+
+    @property
+    def is_editable(self):
+        if self.element.get_attribute("contentEditable") == "true":
+            return True
+        else:
+            return False
+
+    def make_editable(self):
+        if not self.is_editable:
+            self.browser.set_attribute("contentEditable", "true", self)
+
+    def fill(self, value):
+        self.make_editable()
+        self.browser.set_attribute("style", "overflow: visible", self.PARENT_DIV)
+        self.browser.element(self.CLICK_DIV).click()
+        self.element.send_keys(value)
+        return True
```

Comment:
Any way for the fill to fail? I.e. if nothing changes? Just wondering if this should ever return `False`.

Reply:
hm, I am thinking if I could do that when implementing read method Then it will be possible to check that. But in my work on that I've never seen it sending the keys and they are not appearing. It'd fall with "Element no found" or "not intractable" or another selenium exception. I guess it's safe to leave it like that.